core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -68,103 +114,263 @@ public SimpleQueryStringBuilder boost(float boost) {
         this.boost = boost;
         return this;
     }
-    
-    /** Returns the boost of this query. */
+
+    /** Returns the boost to apply to resulting Lucene query.*/
     public float boost() {
         return this.boost;
     }
+    /** Returns the text to parse the query from. */
+    public String text() {
+        return this.queryText;
+    }
 
-    /**
-     * Add a field to run the query against
-     */
+    /** Add a field to run the query against. */
     public SimpleQueryStringBuilder field(String field) {
-        this.fields.put(field, null);
+        if (Strings.isEmpty(field)) {
+            throw new IllegalArgumentException("Supplied field is null or empty.");
+        }
+        this.fieldsAndWeights.put(field, 1.0f);
         return this;
     }
 
-    /**
-     * Add a field to run the query against with a specific boost
-     */
+    /** Add a field to run the query against with a specific boost. */
     public SimpleQueryStringBuilder field(String field, float boost) {
-        this.fields.put(field, boost);
+        if (Strings.isEmpty(field)) {
+            throw new IllegalArgumentException("Supplied field is null or empty.");
```

Comment:
This is a review comment that I got from Shay a long time ago... It may have referred to log lines specifically rather than exception messages. That said, I just  looked at a bunch of IAE's creation and I see we do half and half roughly now.... so maybe we are recently using more uppercase than before. No strong opinions, just would like to have some consistency, which we don't seem to have anyway....